xbmc/addons/gui/GUIDialogAddonInfo.cpp

```diff
@@ -586,98 +584,125 @@ void CGUIDialogAddonInfo::OnEnableDisable()
-        str << " "
-            << StringUtils::Format(g_localizeStrings.Get(39022).c_str(),
-                                   local_addon ? g_localizeStrings.Get(39019).c_str()
-                                               : g_localizeStrings.Get(39018).c_str());
-      else if (it.optional && local_addon)
-        str << " "
-            << StringUtils::Format(g_localizeStrings.Get(39023).c_str(),
-                                   g_localizeStrings.Get(39019).c_str(),
-                                   g_localizeStrings.Get(39018).c_str());
-
-      item->SetLabel2(str.str());
-      item->SetArt("icon", info_addon->Icon());
-      item->SetProperty("addon_id", it.id);
-      items.Add(item);
-    }
-    else
-    {
-      CFileItemPtr item(new CFileItem(it.id));
-      item->SetLabel2(g_localizeStrings.Get(10005)); // Not available
-      items.Add(item);
-    }
-  }
+      // All combinations of depAddon and localAddon validity are possible and information
+      // must be displayed even when there is no depAddon.
+      // info_addon is the add-on to take the information to display (name, icon) from. The
+      // version in the repository is preferred because it might contain more recent data.
 
-  CFileItemPtr backup_item = GetCurrentListItem();
-  while (true)
-  {
-    pDialog->Reset();
-    pDialog->SetHeading(reactivate ? 39024 : 39020);
-    pDialog->SetUseDetails(true);
-    for (auto& it : items)
-      pDialog->Add(*it);
-    pDialog->EnableButton(!reactivate, 186);
-    pDialog->SetButtonFocus(true);
-    pDialog->Open();
-
-    if (pDialog->IsButtonPressed())
-      return true;
+      std::shared_ptr<IAddon> infoAddon = it.m_available ? it.m_available : it.m_installed;
 
-    if (pDialog->IsConfirmed())
-    {
+      if (infoAddon)
+      {
+        if (entryPoint != EntryPoint::UPDATE ||
+            (entryPoint == EntryPoint::UPDATE && !it.m_isInstalledUpToDate()))
+        {
+          const CFileItemPtr item = std::make_shared<CFileItem>(infoAddon->Name());
+          std::stringstream str;
+
+          str << g_localizeStrings.Get(39029).c_str() << ": " << it.m_depInfo.versionMin.asString()
+              << " "; // min: x.y.z
+
+          // dep not installed locally, but it is available from a repo!
+          if (!it.m_installed)
+          {
+            if (entryPoint != EntryPoint::SHOW_DEPENDENCIES)
+            {
+              str << "=> " << g_localizeStrings.Get(24038).c_str() << ": "
+                  << it.m_available->Version().asString(); // => install:
+            }
+          }
+          else // dep is installed locally
+          {
+            str << "/ " << g_localizeStrings.Get(35257).c_str() << ": "
+                << it.m_installed->Version().asString(); // Installed: x.y.z
+
+            if (!it.m_isInstalledUpToDate())
+            {
+              str << " => " << g_localizeStrings.Get(24180).c_str() << ": "
+                  << it.m_available->Version().asString(); // => update to
```

Comment:
About the string usage here maybe not so good.
Them works for our alphabets, but becomes hard on others, where have e.g. right to left.

Maybe to make special own format strings for it?

Reply:
sure, so it would be like `"=> Update to: {0:s}"` for this particular place?

Reply:
Yes, I think so.

Reply:
No assembling of GUI strings in core! No, put string formatting in string.po. Then, each translator can decide what the best format for their language is.

Reply:
moved to `strings.po` with the latest push